src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EmbeddedDataContainerNode.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Represents a node that contains a set of embedded objects. The main function is 
+    /// to serve as a base class, providing symbol name boundaries and node ordering.
+    /// </summary>
+    public abstract class EmbeddedDataContainerNode : ObjectNode
+    {
+        private ObjectAndOffsetSymbolNode _startSymbol;
+        private ObjectAndOffsetSymbolNode _endSymbol;
+
+        public ObjectAndOffsetSymbolNode StartSymbol => _startSymbol;
+        public ObjectAndOffsetSymbolNode EndSymbol => _endSymbol;
+
+        protected EmbeddedDataContainerNode(string startSymbolMangledName, string endSymbolMangledName)
+        {
+            _startSymbol = new ObjectAndOffsetSymbolNode(this, 0, startSymbolMangledName, true);
+            _endSymbol = new ObjectAndOffsetSymbolNode(this, 0, endSymbolMangledName, true);
+        }
+
+        protected override int ClassCode => (int)ObjectNodeOrder.EmbeddedDataContainerNode;
+
+        protected override int CompareToImpl(ObjectNode other, TypeSystemComparer comparer, NameMangler nameMangler)
+        {
+            return StartSymbol.GetMangledName(nameMangler).CompareTo(((EmbeddedDataContainerNode)other).StartSymbol.GetMangledName(nameMangler));
```

Comment:
Actually right now we don't use `IndirectionNode` or `JumpStubNode` in CoreRT (they're TFS only). So the only wrinkle with the current design is the collection nodes. I can avoid the mangler entirely by just saving the `startSymbolMangledName` and comparing them based on that later. That will avoid the need for `NameMangler` in the `CompareToImpl` signature.

I will play with stabilizing the contents of these collections which might need an ISortableSymbolNode or a derived `SortableEmbeddedNode` class.